docs/api.rst

```diff
@@ -96,64 +96,34 @@ base names, such that ``hy.core.macros.foo`` can be called as just ``foo``.
-   new function. In the following example, an anonymous function is defined and
-   passed to another function for filtering output::
-
-       => (setv people [(dict :name "Alice" :age 20)
-       ...              (dict :name "Bob" :age 25)
-       ...              (dict :name "Charlie" :age 50)
-       ...              (dict :name "Dave" :age 5)])
-
-       => (defn display-people [people filter]
-       ...  (for [person people] (if (filter person) (print (:name person)))))
-
-       => (display-people people (fn [person] (< (:age person) 25)))
-       Alice
-       Dave
-
-   Just as in normal function definitions, if the first element of the
-   body is a string, it serves as a docstring. This is useful for giving
-   class methods docstrings::
-
-       => (setv times-three
-       ...   (fn [x]
-       ...    "Multiplies input by three and returns the result."
-       ...    (* x 3)))
-
-   This can be confirmed via Python's built-in ``help`` function::
-
-       => (help times-three)
-       Help on function times_three:
-
-       times_three(x)
-       Multiplies input by three and returns result
-       (END)
+   As :hy:func:`defn`, but no name for the new function is required (or
+   allowed), and the newly created function object is returned. Decorators
+   aren't allowed, either. However, the function body is understood identically
+   to that of :hy:func:`defn`, without any of the restrictions of Python's
+   :py:keyword:`lambda`. See :hy:func:`fn/a` for the asynchronous equivalent.
 
 .. hy:function:: (fn/a [name #* args])
 
-   ``fn/a`` is a variant of ``fn`` than defines an anonymous coroutine.
-   The parameters are similar to ``defn/a``: the first parameter is
-   vector of parameters and the rest is the body of the function. ``fn/a`` returns a
-   new coroutine.
+   As :hy:func:`fn`, but the created function object will be a :ref:`coroutine
+   <py:async def>`.
 
 .. hy:function:: (defn [name #* args])
 
-   Define `name` as a function with `args` as the signature, annotations, and body.
-
-   ``defn`` is used to define functions. It requires two arguments: a name (given
-   as a symbol) and a list of parameters (also given as symbols). Any remaining
-   arguments constitute the body of the function::
+   ``defn`` compiles to a :ref:`function definition <py:function>` (or possibly
+   to an assignment of a :ref:`lambda expression <py:lambda>`). It requires two
+   arguments: a name (given as a symbol; see :hy:func:`fn` for anonymous
+   functions) and a "lambda list", or list of parameters (also given as
+   symbols). Any further arguments constitute the body of the function::
 
        (defn name [params] bodyform1 bodyform2...)
 
-   If there at least two body forms, and the first of them is a string literal,
-   this string becomes the :term:`py:docstring` of the function.
+   An empty body is implicitly ``(return None)``. If there at least two body
```

Comment:
Nitpicking here, not sure if it actually matters, but an empty body actually compiles to `pass`.

Reply:
Right, which is equivalent to Hy's `(return None)`.